app/clients/salesforce/salesforce_client.py

```diff
@@ -98,5 +98,22 @@ def engagement_update_stage(self, service: Service, user: User, stage_name: str)
         """
         session = self.get_session()
         account_id, contact_id = self.contact_update_account_id(session, service, user)
-        salesforce_engagement.update_stage(session, service, stage_name, account_id, contact_id)
+        stage_update = {"StageName": stage_name}
+        salesforce_engagement.update(session, service, stage_update, account_id, contact_id)
+        self.end_session(session)
+
+    def engagement_close(self, service: Service):
```

Comment:
Since we're annotating these we could specify what the function returns as well
```suggestion
    def engagement_close(self, service: Service) -> None:
```